tests/integration_tests/datasources/test_lxd_hotplug.py

```diff
@@ -0,0 +1,121 @@
+import json
+
+import pytest
+
+from cloudinit import safeyaml
+from cloudinit.subp import subp
+from tests.integration_tests.clouds import ImageSpecification
+from tests.integration_tests.decorators import retry
+from tests.integration_tests.instances import IntegrationInstance
+
+USER_DATA = """\
+#cloud-config
+updates:
+  network:
+    when: ["hotplug"]
+"""
+
+UPDATED_NETWORK_CONFIG = """\
+version: 2
+ethernets:
+    eth0:
+        dhcp4: true
+    eth2:
+        dhcp4: true
+"""
+
+
+@retry()
+def ensure_hotplug_exited(client):
+    assert "cloud-init" not in client.execute("ps -A")
+
+
+def get_parent_network(instance_name: str):
+    lxd_network = json.loads(
+        subp("lxc network list --format json".split()).stdout
+    )
+    for net in lxd_network:
+        if net["type"] == "bridge" and net["managed"]:
+            if f"/1.0/instances/{instance_name}" in net.get("used_by", []):
+                return net["name"]
+    return "lxdbr0"
+
+
+@pytest.mark.lxd_container
+@pytest.mark.lxd_vm
+@pytest.mark.user_data(USER_DATA)
+class TestLxdHotplug:
+    @pytest.fixture(autouse=True, scope="class")
+    def class_teardown(self, class_client: IntegrationInstance):
+        # We need a teardown here because on IntegrationInstance teardown,
+        # if KEEP_INSTANCE=True, we grab the instance IP for logging, but
+        # we're currently running into
+        # https://github.com/canonical/pycloudlib/issues/220 .
+        # Once that issue is fixed, we can remove this teardown
+        yield
+        name = class_client.instance.name
+        subp(f"lxc config device remove {name} eth1".split())
+        subp(f"lxc config device remove {name} eth2".split())
+
+    def test_no_network_change_default(
+        self, class_client: IntegrationInstance
+    ):
+        client = class_client
+        assert "eth1" not in client.execute("ip address")
+        pre_netplan = client.read_from_file("/etc/netplan/50-cloud-init.yaml")
+        parent_network = get_parent_network(client.instance.name)
+        assert subp(
+            f"lxc config device add {client.instance.name} eth1 nic name=eth1 "
```

Comment:
This will break in LXD 5.7 due to commit  https://github.com/lxc/lxd/commit/56f29b48c2. 
This is because dnsmasq setups try to give out hostnames to dhcp requesters on managed networks. Each of thes now has a strict requirement to only allow on such device attached to the same hostname on that network so there aren't DNS collisions for mutliple hosts.


I think our solution here is to create a separate bridge network first for this integration test.
f`lxc network create {client.instance.name}-net --type=bridge ipv4.address=10.10.40.1/24 ipv4.nat="true"` and then attaching to that network. I'm confirming this right now.

Reply:
But again, let's avoid this until we have a discussion with lxd folks next week on the appropriate expectations w/ managed bridges.